[Doc] Add a section on StarRocks alert rules and their handling methods (#51662)
StarRocks/starrocks: docs/en/administration/management/monitoring/alert.md

# StarRocks cluster alarm rules and handling methods

This article introduces the alarm items that need attention and their handling methods from multiple dimensions such as business continuity, cluster availability, and machines.

> The following $job_name needs to be replaced with the corresponding job name in the prometheus configuration, and $fe_leader needs to be replaced with the corresponding ip of the leader fe.

Reviewer: ```suggestion
The following `$job_name` needs to be replaced with the corresponding job name in the prometheus configuration, and `$fe_leader` needs to be replaced with the corresponding IP of the leader FE.
```

## Service pending alarm

### FE service suspended

Expression:

```sql
count(up{group="fe", job="$job_name"}) >= 3
```

Alarm description:

> Please adjust the number of surviving Fe nodes according to the actual number of Fe nodes.

Reviewer: ```suggestion
Please adjust the number of surviving FE nodes according to the actual number of FE nodes.
```

Processing method:

FE service is down, try to pull it up.

### BE service suspended

Expression:

```sql
node_info{type="be_node_num", job="$job_name",state="dead"} > 1
```

Alarm description:

The number of pending nodes are greater than 1.

Processing method:

> There is a be service down, try to pull up.

Reviewer: ```suggestion
There is a BE service down, try to pull up.
```

## Machine overload alarm

### BE cpu alarm

Expression:

```sql
(1-(sum(rate(starocks_be_cpu{mode="idle", job="$job_name",instance=~".*"}[5m])) by (job, instance)) / (sum(rate(starocks_be_cpu{job="$job_name",host=~".*"}[5m])) by (job, instance))) * 100 > 90
```

Alarm description:

CPU util exceeds 90%

Processing method:

There may be large queries or a large amount of data import. First, you can get the top result and take a screenshot.

```shell
top -Hp $be_pid
```

Get the result of the perf and send it to StarRocks support for positioning.

```shell
sudo perf top -p $be_pid -g >/tmp/perf.txt #Execute for 1-2 minutes, then cancel with CTRL+C.
```

In case of emergency, in order to restore the service as soon as possible, you can try to restart the corresponding BE service (keep the stack before restarting) (emergency: BE node process monitoring continues to be abnormal and full, unable to reduce CPU usage through effective positioning measures).

### Memory alarm

Expression:

```sql
(1-node_memory_MemAvailable_bytes{instance=~".*"}/node_memory_MemTotal_bytes{instance=~".*"})*100 > 90
```

Alarm description:

Memory usage rate exceeds 90%.

Processing method:

> Troubleshooting methods can refer to [memory problem troubleshooting](https://docs.starrocks.io/docs/administration/management/resource_management/Memory_management/) or get [heap profile](https://github.com/StarRocks/starrocks/pull/35322) troubleshooting.

Reviewer: ```suggestion
Troubleshooting methods can refer to [memory problem troubleshooting](./management/resource_management/Memory_management.md) or get [heap profile](https://github.com/StarRocks/starrocks/pull/35322) troubleshooting.
```

Reviewer: We need to copy the troubleshooting from that PR into the docs. @DanJing-coder can we add it in this doc, or should it be somewhere else?

- In case of emergency, in order to restore the service as soon as possible, you can try to restart the corresponding BE service (emergency: BE node process memory monitoring continues to be abnormal and full, and it is impossible to reduce memory usage through effective positioning).
- If other services of the mixed department affect SR, other services can be considered to be stopped in an emergency

### Disk alarm

#### Disk load alarm

Expression:

```sql
rate(node_disk_io_time_seconds_total{instance=~".*"}[1m]) * 100 > 90
```

Alarm description:

Disk load over 90%

Processing method:

If there is a node_disk_io_time_seconds_total alarm, then first confirm whether there is a change in the business, if there is, whether the change can be rolled back, maintain the resource balance conditions before, if not, consider whether it is normal business growth leads to the need to expand resources.

> You can use the iotop tool to analyze and monitor disk I/O usage. Iotop has a UI similar to top, including pid, user, I/O, process and other related information.

Reviewer: ```suggestion
You can use the `iotop` tool to analyze and monitor disk I/O usage. Iotop has a UI similar to top, including pid, user, I/O, process and other related information.
```
